### PR TITLE
services into drawer (DEV-781)

### PR DESCRIPTION
**Pull request by Davit-BetterAngels**
DEV-781

## Summary by Sourcery

Refactor the handling of requested and provided services by introducing a new RequestedProvidedServices component and a ServicesModal for managing service selection. This change simplifies the codebase by consolidating service-related logic and UI elements, improving maintainability and user experience.

Enhancements:
- Refactor the RequestedServices and ProvidedServices components to use a new RequestedProvidedServices component for handling service requests.
- Introduce a ServicesModal component to manage the selection and submission of services in a modal interface.
- Add a new constants file to define service categories and their respective items for better organization and maintainability.

### CODE REVIEW

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx:57]: I think we may have this issue in other places, but when clicking this the keyboard hides the input.

Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx:57]: fixed

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx:28]: this seems to `append` service instead of `toggle`?

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx:31]: maybe call it `handleAddOtherService`?
"legal", "transportation" and "other" seem to be categories, and we are adding a "service"

Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx:31]: could we rename `e` to service, serviceName or newService or something more descriptive?
think that would make line below more clear

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:299]: would love to get away from these 300+ loc files

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:238]: I think this should be renamed a bit. By the time I get to this line I'm pretty confused and not sure what `services` means. Maybe renaming `Services` - the Services you map over - to `Categories` (categories have services) may help, assuming i'm following this correctly

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:239]: perhaps can call it `setSelectedServices`?

File: libs/expo/betterangels/src/lib/static/constants.ts
Comment (tglaz) [inline libs/expo/betterangels/src/lib/static/constants.ts:72]: on my phone this one doesn't fit, so may need to wrap it

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:192]: to be consistent I'd suggest user either `.filter(item) and .map(item)` in both or `service` in both as they're same objects

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:182]: seems like empty string shouldn't be allowed?

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:165]: oh many, now I know what you meant earlier. 
Could potentially be 25+ separate requests 🤣

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx:32]: I know we use the pattern of passing the `setServices` (or setWhatever) function down to children, but seems like a lot of responsibility to give to a checkbox. Any reason it can't return a boolean and the parent take care of the updating the actual services list?

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (mikefeldberg) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:165]: > potentially be 25+ separate requests
😳

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (tglaz) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:105]: wait, so `sourcery-ai[bot]` gave both a thumbs UP and DOWN?

File: apps/betterangels/src/app/(private-screens)/add-note/ProvidedServices/index.tsx
Comment (tglaz) [reply apps/betterangels/src/app/(private-screens)/add-note/ProvidedServices/index.tsx:23]: Ok. So I totally agree with this one . But I still wouldn't change it just on principle.

File: libs/expo/betterangels/src/lib/static/constants.ts
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/static/constants.ts:4]: renamed

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx:47]: leftover fromo previous flow. removed

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx:31]: renamed to newService

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/OtherCategory.tsx:28]: renamed

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:238]: as I renamed all others I think we can keep "services" for the state

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:239]: as I left state name "services" I will keep setServices

File: libs/expo/betterangels/src/lib/static/constants.ts
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/static/constants.ts:11]: changed and removed title

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:192]: renamed to serviceItem both

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:182]: changed to null

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx:39]: didn't get this part.

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx:32]: as service checkbox is not reusable for other cases I didn't want to make parent bigger.

File: libs/expo/betterangels/src/lib/static/constants.ts
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/static/constants.ts:72]: fixed

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:170]: done

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx
Comment (tglaz) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx:37]: what I meant by `return early` was to 

```javascript
const handleCheck = () => {
  if (isChecked) {
    setServices([...]);

    return;
  } 
  
  if (serviceEntry) {
    setServices([...]);

    return;
  }
  
  setServices([
    ...services,
    { enum: service, id: undefined, markedForDeletion: false },
  ]);
};
```

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx
Comment (tglaz) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx:247]: So I'll leave it up to you, but here's a version to demonstrate what i mean.
I'm sure it's much more clear to you as you've super familiar with this code, but as I come in to read it's a bit of an effort to follow.

```javascript
{ServicesByCategory.map((category) => ( // we map over categories
  <View key={category.title}> // category has a title
    <TextBold mb="xs">{category.title}</TextBold>
    {category.services.map((service: ServiceEnum, idx) => { // category has services
      return (
        <ServiceCheckbox
          key={service}
          services={services} // which services?
          setServices={setServices} // which services are we setting?
          noteId={noteId} // not sure we need this
          service={service}
          idx={idx}
        />
      );
    })}
  </View>
))}
```

File: libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx
Comment (tglaz) [reply libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx:30]: ok. 
won't block on this, but I think that if parent component is too big then probably better to address that directly as opposed to moving responsibilities down to children.

**Review by tglaz — APPROVED** [review]

**Pull request merged**
